lib/utils.sh

```diff
@@ -84,7 +84,7 @@ resolve_legacy_version() {
   case "$strategy" in
   latest_installed)
     _list() {
-      ASDF_NODEJS_SKIP_NODEBUILD_UPDATE=1 list_installed_versions nodejs
+      ASDF_NODEJS_SKIP_NODEBUILD_UPDATE=1 list_installed_versions nodejs | sort -V
```

Comment:
I don't think `sort -V` is available on older versions of MacOS. In asdf we had to do this: https://github.com/asdf-vm/asdf/blob/ccdd47df9b73d0a22235eb06ad4c48eb57360832/lib/commands/command-update.bash#L45-L49

Reply:
> I don't think `sort -V` is available on older versions of MacOS. In asdf we had to do this: https://github.com/asdf-vm/asdf/blob/ccdd47df9b73d0a22235eb06ad4c48eb57360832/lib/commands/command-update.bash#L45-L49

This could be used instead. I copied `sort_versions` into this PR.

As I already noted in #358, asdf itself [wants coreutils to be installed on macOS](https://asdf-vm.com/guide/getting-started.html#_1-install-dependencies), but I guess that does not include the requirement that GNU `sort` should be first in `$PATH`.